de.metas.business/src/main/java/org/adempiere/mm/attributes/api/IAttributeDAO.java

```diff
@@ -190,4 +189,14 @@
 	 * @return true if given attribute is expected to have a huge amount of {@link I_M_AttributeValue}s.
 	 */
 	boolean isHighVolumeValuesList(I_M_Attribute attribute);
+
+	/**
+	 * Retrieve an M_AttributeValue entry by it's attribute and Name.
+	 * Only use if you are sure there aren't several attribute values with the same name!
```

Comment:
Since you use this method, how do you make sure there aren't several attribute values with the same name ;-) ?

AFAIU, since every admin-user might add attribute values, a dev calling this method can never be sure, unless he/she writes it's own list-retrieval method and checks there retunred lists's size, before calling your method ;-).
Checking a large client's DB, i found that there are in fact dublicate names.
Also note that the attribute values can have not-0 AD_Client_IDs and AD_Org_IDs.

So i would propose to 
* add a UC constraint on isactive='Y' that also includes attribute-ID, name, org-ID and client-ID
now, and we would clean up the doubles from their DB
* also provide the current ctx to the method, and return the first match (preferning AD_Org_ID>0, AD_Client_ID>0 etc)